pkg/inframetadata/internal/hostmap/hostmap.go

```diff
@@ -43,10 +43,18 @@ func strField(m pcommon.Map, key string) (string, bool, error) {
 		// Field not available, don't update but don't fail either
 		return "", false, nil
 	}
-	if val.Type() != pcommon.ValueTypeStr {
+
+	var value string
+	switch val.Type() {
+	case pcommon.ValueTypeStr:
+		value = val.Str()
+	case pcommon.ValueTypeInt:
```

Comment:
Maybe you should update the documentation (or name) of this function to indicate that it also returns an int field now.

Reply:
Makes sense, fixed!